command/agent/config/config.go

```diff
@@ -254,69 +481,29 @@ func LoadConfig(path string) (*Config, error) {
 		return nil, fmt.Errorf("error parsing 'template': %w", err)
 	}
 
-	if result.Cache != nil && result.APIProxy == nil && len(result.Listeners) > 0 {
+	if result.Cache != nil && result.APIProxy == nil {
 		result.APIProxy = &APIProxy{
 			UseAutoAuthToken:   result.Cache.UseAutoAuthToken,
 			ForceAutoAuthToken: result.Cache.ForceAutoAuthToken,
 		}
 	}
 
-	if result.Cache != nil {
-		if len(result.Listeners) < 1 && len(result.Templates) < 1 {
-			return nil, fmt.Errorf("enabling the cache requires at least 1 template or 1 listener to be defined")
-		}
-
-		if result.Cache.UseAutoAuthToken {
-			if result.AutoAuth == nil {
-				return nil, fmt.Errorf("cache.use_auto_auth_token is true but auto_auth not configured")
-			}
-			if result.AutoAuth != nil && result.AutoAuth.Method != nil && result.AutoAuth.Method.WrapTTL > 0 {
-				return nil, fmt.Errorf("cache.use_auto_auth_token is true and auto_auth uses wrapping")
-			}
-		}
-	}
-
-	if result.APIProxy != nil {
-		if len(result.Listeners) < 1 {
-			return nil, fmt.Errorf("configuring the api_proxy requires at least 1 listener to be defined")
-		}
-
-		if result.APIProxy.UseAutoAuthToken {
-			if result.AutoAuth == nil {
-				return nil, fmt.Errorf("api_proxy.use_auto_auth_token is true but auto_auth not configured")
-			}
-			if result.AutoAuth != nil && result.AutoAuth.Method != nil && result.AutoAuth.Method.WrapTTL > 0 {
-				return nil, fmt.Errorf("api_proxy.use_auto_auth_token is true and auto_auth uses wrapping")
-			}
-		}
-	}
-
-	if result.AutoAuth != nil {
-		if len(result.AutoAuth.Sinks) == 0 &&
-			(result.APIProxy == nil || !result.APIProxy.UseAutoAuthToken) &&
-			len(result.Templates) == 0 {
-			return nil, fmt.Errorf("auto_auth requires at least one sink or at least one template or api_proxy.use_auto_auth_token=true")
-		}
-	}
-
 	err = parseVault(result, list)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing 'vault':%w", err)
 	}
 
-	if result.Vault == nil {
```

Comment:
We no longer set a default Vault config retries, we simply use the default if it's not set.